src/rsa.rs

```diff
@@ -475,6 +486,51 @@ extern {
                                  rng: *mut rand::RAND) -> c::int;
 }
 
+#[cfg(feature = "rsa_signing")]
+#[allow(non_snake_case)]
+#[doc(hidden)]
+#[no_mangle]
+pub unsafe extern fn BN_rand_range(dest: *mut BIGNUM, range: *const BIGNUM,
+                                   rng: *mut rand::RAND) -> c::int {
```

Comment:
> C) We start by writing a new key-generation function, then merge them later, instead of refactoring first. My gut reaction is against that. What's the advantage?

One of my concerns is that we can't use the same strategy of stack-allocated oversized buffers for RSA blinding that we use for ECC. We do it for ECC because the objects in ECC are all roughly the same size (+/- 16 bytes). In RSA, the difference between the smallest `BIGNUM` (1024 bits) and largest `BIGNUM` (8192 bits) is extreme. And, the largest `BIGNUM`s we have to support are very rarely used. So, if we make everything big enough to accomodate the large keys, then we'll end up wasting quite a bit of memory, because RSA private keys consist of many such values, and often there are many blindings in memory at once as well. So, I'm hoping to find a solution for this particular PR that doesn't require us to add heap allocations in the Rust code and which doesn't require the large static buffers.

Also, I'm hoping we can, at least, find some convenient way to break up the PR into smaller commits.